client/src/components/Booker.js

```diff
@@ -60,36 +61,82 @@ class Booker extends React.Component {
       })
    }
 
-   handleChange = (date) => {
-    this.setState({
-      startDate: date
-    });
-  }
 
-  addMonths = (today, monthsToAdd) => {
+   addMonths = (today, monthsToAdd) => {
+
+    const maxDate = moment(today)
+
+    return maxDate.add(monthsToAdd, 'months')._d
+    }
+
+
+  // Triggerts handleChange for start date
+  handleChangeStart = (startDate) => {
+
+    console.log('startDate', startDate)
+
+    startDate = startDate || this.state.startDate;
+
+    let endDate = this.state.endDate;
+
+    if(moment(startDate).isAfter(endDate)){
+      endDate = startDate
+      this.setState({endDate})
+    }
+
+    this.setState({startDate})
+  
+  };
+
+  // Triggers handleChage for end date
+  handleChangeEnd = (endDate) => {
+
+    console.log('endDate', endDate)
+
+    endDate = endDate || this.state.endDate;
 
-      const maxDate = moment(today)
+    const startDate = this.state.startDate
+
+    if(moment(startDate).isAfter(endDate)){
+      console.log('true')
+      endDate = startDate
+    }
+
+    this.setState({endDate})
 
-      return maxDate.add(monthsToAdd, 'months')._d
+  }
+
+  componentDidMount(){
+    this.handleGetBookedDates()
   }
 
   render(){
     return(
+
       <div>
       <DatePicker
-        selected={this.state.startDate}
-        minDate={new Date()}
-        maxDate={this.addMonths(new Date(), 5)}
-        onChange={this.handleChange}
-        excludeDates={this.state.bookedDates}
+          selected={this.state.startDate}
+          selectsStart
+          excludeDates={this.state.bookedDates}
+          minDate={new Date()}
+          maxDate={this.addMonths(new Date(), 5)}
+          startDate={this.state.startDate}
+          endDate={this.state.endDate}
+          onChange={this.handleChangeStart}
       />
 
-      <p>{this.state.message}</p>
+      <DatePicker
+          selected={this.state.endDate}
+          selectsEnd
+          excludeDates={this.state.bookedDates}
+          minDate={new Date()}
+          maxDate={this.addMonths(new Date(), 5)}
+          startDate={this.state.startDate}
+          endDate={this.state.endDate}
+          onChange={this.handleChangeEnd}
```

Comment:
same question as before, i'm used to binding functions

Reply:
I like to bind them via arrow functions